src/test/java/codeu/controller/ChatServletTest.java

```diff
@@ -178,37 +185,72 @@ public void testDoPost_StoresMessage() throws IOException, ServletException {
+                      "$2a$10$eDhncK/4cNH2KE.Y51AWpeL8/5znNBQLuAFlyJpSYNODR/SJQ/Fg6",
+                      Instant.now());
+      Mockito.when(mockUserStore.getUser("test_username")).thenReturn(fakeUser);
 
-    Conversation fakeConversation =
-        new Conversation(UUID.randomUUID(), UUID.randomUUID(), "test_conversation", Instant.now());
-    Mockito.when(mockConversationStore.getConversationWithTitle("test_conversation"))
-        .thenReturn(fakeConversation);
+      Conversation fakeConversation =
+              new Conversation(UUID.randomUUID(), UUID.randomUUID(), "test_conversation", Instant.now());
+      Mockito.when(mockConversationStore.getConversationWithTitle("test_conversation"))
+              .thenReturn(fakeConversation);
 
-    Mockito.when(mockRequest.getParameter("message"))
-        .thenReturn("Contains <b>html</b> and <script>JavaScript</script> content.");
+      Mockito.when(mockRequest.getParameter("message"))
+              .thenReturn("Contains <b>html</b> and <script>JavaScript</script> content.");
 
-    chatServlet.doPost(mockRequest, mockResponse);
+      chatServlet.doPost(mockRequest, mockResponse);
 
-    ArgumentCaptor<Message> messageArgumentCaptor = ArgumentCaptor.forClass(Message.class);
-    Mockito.verify(mockMessageStore).addMessage(messageArgumentCaptor.capture());
-    Assert.assertEquals(
-        "Contains html and  content.", messageArgumentCaptor.getValue().getContent());
+      ArgumentCaptor<Message> messageArgumentCaptor = ArgumentCaptor.forClass(Message.class);
+      Mockito.verify(mockMessageStore).addMessage(messageArgumentCaptor.capture());
+      Assert.assertEquals(
+              "Contains html and  content.", messageArgumentCaptor.getValue().getContent());
 
-    Mockito.verify(mockResponse).sendRedirect("/chat/test_conversation");
+      Mockito.verify(mockResponse).sendRedirect("/chat/test_conversation");
   }
+
+    @Test
+    public void testDoPost_AddsActivity() throws IOException, ServletException {
+        Mockito.when(mockRequest.getRequestURI()).thenReturn("/chat/test_conversation");
+        Mockito.when(mockSession.getAttribute("user")).thenReturn("test_username");
+
+        User fakeUser =
+                new User(
+                        UUID.randomUUID(),
+                        "test_username",
+                        "$2a$10$bBiLUAVmUFK6Iwg5rmpBUOIBW6rIMhU1eKfi3KR60V9UXaYTwPfHy",
+                        Instant.now());
+        Mockito.when(mockUserStore.getUser("test_username")).thenReturn(fakeUser);
+
+        Conversation fakeConversation =
+                new Conversation(UUID.randomUUID(), UUID.randomUUID(), "test_conversation", Instant.now());
+        Mockito.when(mockConversationStore.getConversationWithTitle("test_conversation"))
+                .thenReturn(fakeConversation);
+
+        Mockito.when(mockRequest.getParameter("message")).thenReturn("Test message.");
+
+        chatServlet.doPost(mockRequest, mockResponse);
+
```

Comment:
All the above in this function is just mocking really; the actually asserts start here